django/db/backends/base/schema.py

```diff
@@ -1063,9 +1063,25 @@ def _field_should_be_altered(self, old_field, new_field):
         _, new_path, new_args, new_kwargs = new_field.deconstruct()
         # Don't alter when:
         # - changing only a field name
+        # - changing an attribute that doesn't affect the schema
         # - adding only a db_column and the column name is not changed
-        old_kwargs.pop('db_column', None)
-        new_kwargs.pop('db_column', None)
+        non_database_fields = [
```

Comment:
I don't think this is the best way to address this issue. You will always have to keep this list of fields updated. Instead we could change `_alter_field` field to do nothing when the constraint hasn't changed. We could just compare the old and new field. 

This is the `if` that drops the constraint.
https://github.com/django/django/blob/master/django/db/backends/base/schema.py#L590

and the one that adds it back
https://github.com/django/django/blob/master/django/db/backends/base/schema.py#L810

Reply:
This is not only about constraints but also about noop `ALTER FIELD` operations. Field alteration can cause many DDL changes.

> We could just compare the old and new field.

That's exactly what we're doing here, we compare fields but without attributes that don't impact schema changes.